WordPress/Classes/Models/Notifications/Notification.swift

```diff
@@ -412,3 +411,21 @@ extension Notification: Notifiable {
         return notificationId
     }
 }
+
+private class NotificationCachedAttributesObserver: NSObject {
+    override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey: Any]?, context: UnsafeMutableRawPointer?) {
+        guard let keyPath, let notification = object as? Notification, Notification.cachedAttributes.contains(keyPath) else {
+            return
+        }
+
+        // Note:
+        // Cached Attributes are only consumed on the main thread, when initializing UI elements.
+        // As an optimization, we'll only reset those attributes when we're running on the main thread.
+        //
+        guard notification.managedObjectContext?.concurrencyType == .mainQueueConcurrencyType else {
+            return
+        }
```

Comment:
The original observer logic is kept, including this one here. But I think not invaliding the cache just because the notification object is in a background context is problematic, see [the unit test  below](https://github.com/wordpress-mobile/WordPress-iOS/pull/19395/files#diff-b159138f3547f7e8daa93740188dd5d1432f1314e123260fefedfa94ae8d99abR244) for more.